### PR TITLE
CASMINST-5855 - Fix UnboundLocalError in cubs_tool.py

### PR DESCRIPTION
**Pull request by cdelatte-hpe**
# Description

When we enter our in_family check function, a variable was not getting set properly causing the noted error.

Since this was adding the whole script I put comments on what changed in comparison to the one that would be in the image.  

# Checklist Before Merging

- [ ] If I added any command snippets, the steps they belong to follow the prompt conventions (see [example][1]).
- [ ] If I added a new directory, I also updated `.github/CODEOWNERS` with the corresponding team in [Cray-HPE][2].
- [ ] My commits or Pull-Request Title contain my JIRA information, or I do not have a JIRA.

[1]: https://github.com/Cray-HPE/docs-csm/blob/main/introduction/documentation_conventions.md#using-prompts
[2]: https://github.com/Cray-HPE/teams

### CODE REVIEW

File: scripts/cubs_tool.py [hunk 1]
```diff
@@ -0,0 +1,421 @@
...
+    #image_present = image_exists.returncode
+    if image_status != 0:
+        try:
+            subprocess.run(["podman", "pull", remote_registry + "/ceph/ceph:" + new_vers])
+            image_present = subprocess.run(["podman", "image", "exists", remote_registry + "/ceph/ceph:" + new_vers], check=True)
+        except:
+            print("The image cannot be pulled from the registry.  Please ensure you have the right image path and version.")
+            exit(2)
+    registry_vers = json.loads(subprocess.check_output(["podman", "search", remote_registry + "/ceph/ceph", "--list-tags", "--format=json"]))[0]['Tags']
+    upgrade_available = False
```
Comment (cdelatte-hpe) [inline scripts/cubs_tool.py:185]: added here to set value as default.

File: scripts/cubs_tool.py [hunk 2]
```diff
@@ -0,0 +1,421 @@
...
+        elif new_vers in registry_vers:
+            upgrade_available = True
+            if not quiet:
+              print ("Upgrade Available!!  The specified version %s has been found in the registry" % (new_vers))
+    elif in_family_override:
+        if version.parse(new_vers) == version.parse(current_version):
+            upgrade_available = True
+            if not quiet:
+                print ("Upgrade Available!!  The specified version %s has been found in the registry" % (new_vers))
+        elif version.parse(new_vers) != version.parse(current_version):
```
Comment (cdelatte-hpe) [inline scripts/cubs_tool.py:199]: added this logic in so they know why it failed when not run in quiet mode

**Review by bklei — APPROVED** [review]

File: scripts/cubs_tool.py [hunk 3]
```diff
@@ -0,0 +1,421 @@
...
+def fetch_base_current_vers():
+    cmd = {"prefix":"version", "format":"json"}
+    cmd_results = cluster.mon_command(json.dumps(cmd), b'', timeout=5)
+    results = json.loads(cmd_results[1])
+    for key,value in results.items():
+      current_version = str(value.split(' ')[2])
+    return current_version
+
+def fetch_curr_sha(cmd, new_version):
+    #cmd = {"prefix":"orch ps", "format":"json"}
```
Comment (kburns-hpe) [inline scripts/cubs_tool.py:88]: Can you remove this comment? it's a bit confusing.

**Review by kburns-hpe — APPROVED** [review]

File: scripts/cubs_tool.py (hunk 3 above)
Comment (cdelatte-hpe) [reply scripts/cubs_tool.py]: removed

**Review by rustydb — APPROVED** [review]

**Pull request merged**